### PR TITLE
Scheduler support

### PR DESCRIPTION
**Pull request by RichardKelley**
Adds support for random sampling and file-based match schedules.

### CODE REVIEW

File: lm_tournament_eval/__main__.py [hunk 1]
```diff
@@ -133,10 +142,40 @@ def run_tournament():
...
+    if args.file_schedule is not None and args.sampling_schedule is not None:
+        logging.error("Cannot set file_schedule and sampling_schedule at same time.")
+        sys.exit(1)
+
+    if args.file_schedule is not None:
+        logging.info("Using {args.file_schedule} for match schedule.")
+        scheduler = FileScheduler(args.file_schedule)
+    elif args.sampling_schedule is not None and args.sampling_schedule:
+
+        if args.match_size is None:
```
Comment (Duncanswilson) [inline lm_tournament_eval/__main__.py]: both `match_size` and `num_rounds` default to `1` so these error conditionals are not needed

Comment (RichardKelley) [reply lm_tournament_eval/__main__.py]: Yup. Removing those conditionals now.

File: lm_tournament_eval/api/tournament.py [hunk 2]
```diff
@@ -201,56 +209,73 @@ def run_tournament(self):
...
+                                                    eval_tasks=eval_tasks,
+                                                    task_dict=task_dict,
+                                                    padding_requests=padding_reqests1,
+                                                    batch_size=self.config.batch_size,
+                                                    device=self.config.device,
+                                                    limit=self.config.limit
+                                                )
+
+                rounds_per_task = []
+                match_results = {}
```
Comment (Duncanswilson) [inline lm_tournament_eval/api/tournament.py:269]: we're not using `match_results` anywhere but shouldn't it's creation get moved up? or else it is reset every time between tasks?

**Review by Duncanswilson — COMMENTED** [review]
two small comments but everything looks good to me. I'm going to test multi-gpu locally now and will comment again once that checks out

**Review by Duncanswilson — APPROVED** [review]
both model and data parallelism work still, so this PR is ready to merge - I think we can leave the `match_results` stuff alone, since we're moving to the DB soon.

**Pull request merged**